### PR TITLE
fix(web-client): Fix disable submit button.

### PR DESCRIPTION
**Pull request by IscoRuta98**
**Bug**: When user gets incorrect answers to security questions, and wants to try again, the submit button has been disabled.
**Fix**: Submit button is disabled when answers are checked, but enabled when user wants to submit attempts to security questions.

**Bug**: User can enter an empty PIN as their new PIN.
**Fix**: Add validation to make sure new PIN is valid.

### CODE REVIEW

**Comment by jongbonga** [comment]
Remove the logs please

File: web-client/src/app/views/pin-reset/pin-reset.page.ts [hunk 1]
```diff
@@ -52,27 +54,47 @@ export class PinResetPage implements OnInit {
...
+            input: 'password',
+            inputPlaceholder: 'Enter your PIN here',
+            inputAttributes: {
+              autocomplete: 'off',
+              autocapitalize: 'off',
+              autocorrect: 'off',
+            },
+            preConfirm: (pin) => {
+              // Perform additional PIN verification here
+              if (!/^\d+$/.test(pin)) {
```
[the remark this answers is not in the document]
Comment (jongbonga) [reply web-client/src/app/views/pin-reset/pin-reset.page.ts:77]: why not use [isNAN()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN)?

[the remark this answers is not in the document]
Comment (IscoRuta98) [reply web-client/src/app/views/pin-reset/pin-reset.page.ts]: I have used regular expressions before, hence it was familiar to me, however the above `isNan` function is much more efficient in this example, hence I've changed to using it.

File: web-client/src/app/views/pin-reset/pin-reset.page.ts [hunk 2]
```diff
@@ -52,27 +54,47 @@ export class PinResetPage implements OnInit {
...
-          inputAttributes: {
-            autocomplete: 'off',
-            minlength: '4',
-            autocapitalize: 'off',
-            autocorrect: 'off',
-          },
-        });
+        const { value: password, dismiss: cancelReset } =
+          await this.notification.swal.fire({
+            title: 'Enter new PIN.',
```
[the remark this answers is not in the document]
Comment (IscoRuta98) [reply web-client/src/app/views/pin-reset/pin-reset.page.ts]: Resolved.

File: web-client/src/app/views/pin-reset/pin-reset.page.ts [hunk 3]
```diff
@@ -52,27 +54,47 @@ export class PinResetPage implements OnInit {
...
+            input: 'password',
+            inputPlaceholder: 'Enter your PIN here',
+            inputAttributes: {
+              autocomplete: 'off',
+              autocapitalize: 'off',
```
[the remark this answers is not in the document]
Comment (IscoRuta98) [reply web-client/src/app/views/pin-reset/pin-reset.page.ts]: Resolved.

File: web-client/src/app/views/pin-reset/pin-reset.page.ts [hunk 4]
```diff
@@ -87,36 +109,33 @@ export class PinResetPage implements OnInit {
...
             title: 'Unexpected Failure',
             text: 'An unexpected error occured. Please try again.',
           });
           this.navCtrl.navigateRoot('/');
+          this.isBusySaving = false;
         }
       } catch (err) {
         console.log('catch err: ' + err);
       }
     } else if ('NotFound' in initialResult) {
```
[the remark this answers is not in the document]
Comment (IscoRuta98) [reply web-client/src/app/views/pin-reset/pin-reset.page.ts:120]: Resolved.

File: web-client/src/app/views/pin-reset/pin-reset.page.ts [hunk 5]
```diff
@@ -52,27 +54,45 @@ export class PinResetPage implements OnInit {
     );
 
     if ('InvalidAuth' in initialResult) {
-      console.log(initialResult.InvalidAuth);
       this.notification.swal.fire({
         icon: 'warning',
         title: 'Incorrect Answers!',
         text: 'Authentication failed, please ensure that the answers to the security questions are correct.',
       });
       this.navCtrl.navigateRoot('/pin-reset');
```
Comment (jongbonga) [inline web-client/src/app/views/pin-reset/pin-reset.page.ts:62]: are you not already in this page when this happens?

**Review by jongbonga — COMMENTED** [review]
`onSubmit` function is quite big... can you break it down to smaller functions?

**Comment by IscoRuta98** [comment]
> `onSubmit` function is quite big... can you break it down to smaller functions?

@jongbonga, I've broken down the onSubmit() funciton to smaller functions.

File: web-client/src/app/utils/notification/swal-helper.ts [hunk 6]
```diff
@@ -26,6 +26,22 @@ export class SwalHelper {
     });
   }
 
+  showIncorrectSecutiyAnswers() {
+    this.swal.fire({
+      icon: 'warning',
+      title: 'Incorrect Answers!',
```
Comment (jongbonga) [inline web-client/src/app/utils/notification/swal-helper.ts:32]: replace title with titleText please

File: web-client/src/app/utils/notification/swal-helper.ts [hunk 7]
```diff
@@ -26,6 +26,22 @@ export class SwalHelper {
...
+      text: 'Authentication failed, please ensure that the answers to the security questions are correct.',
+    });
+  }
+
+  showPinResetSuccessNotification() {
+    this.swal.fire({
+      icon: 'success',
+      title: 'Pin Reset Successfully!',
```
Comment (jongbonga) [inline web-client/src/app/utils/notification/swal-helper.ts:40]:
```suggestion
      titleText: 'Pin Reset Successfully!',
```

File: web-client/src/app/utils/notification/swal-helper.ts (hunk 6 above)
Comment (jongbonga) [inline web-client/src/app/utils/notification/swal-helper.ts:32]:
```suggestion
      titleText: 'Incorrect Answers!',
```

**Review by jongbonga — CHANGES_REQUESTED** [review]

File: web-client/src/app/views/pin-reset/pin-reset.page.ts [hunk 8]
```diff
@@ -41,45 +41,47 @@ export class PinResetPage implements OnInit {
...
+            if ('Reset' in pinResetResult) {
+              this.notification.showPinResetSuccessNotification();
+              this.navCtrl.navigateRoot('/');
+            } else {
+              this.notification.showUnexpectedFailureWarning();
+              this.navCtrl.navigateRoot('/');
+            }
           }
+        } catch (err) {
+          console.log('catch err: ' + err);
```
Comment (jongbonga) [inline web-client/src/app/views/pin-reset/pin-reset.page.ts:77]: Do we do nothing on error?

Comment (IscoRuta98) [reply web-client/src/app/views/pin-reset/pin-reset.page.ts:77]: Apart from logging the error, not really. Perhaps adding a popup notification displaying "An unexpected error".

**Review by jongbonga — APPROVED** [review]

**Pull request merged**